.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: "System Smoke Tests"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:    
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  setup:
+    runs-on: self-hosted
+    steps:
+    - name: Check Test Cluster
+      run: |
+        if kubectl get nodes; then
+          kubectl get nodes
+        else
+          echo "test cluster not active. Attempting to create through kind..."
+          kind delete cluster 2>/dev/null
+          kind create cluster
+          kubectl get nodes
+        fi
+  test-source:
+    runs-on: self-hosted
+    needs: setup
+    steps:
+    - uses: actions/checkout@v2
+    - name: Clean Cluster
+      run: |
+        kubectl delete all --all --wait=true --force --grace-period=0
+        sleep 5
+    - name: Deploy From Source
+      run: |
+        skaffold run
+    - name: Wait For Pods
+      run: |
+        sleep 60
+        kubectl get pods
+        echo waiting for ready state...
+        kubectl wait --for=condition=ready pod -l app=adservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=cartservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=checkoutservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=currencyservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=emailservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=frontend --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=loadgenerator --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=paymentservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=productcatalogservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=recommendationservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=shippingservice --timeout=500s
+    - name: Test HTTP
+      timeout-minutes: 5
+      run: |
+        RESULT=" "
+        while [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; do
+          sleep 1
+          RESULT=$(kubectl exec deployments/frontend -- sh -c "wget --spider -S "http://frontend" 2>&1 | grep 'HTTP/'")
+          echo "front end response: $RESULT"
+        done
+        if [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; then 
+          exit 1
+        fi
+  test-release:
```

Comment:
tbh I am not sure about the value of testing `release/` on every commit

Reply:
also if you eliminate this, you will be avoiding duplication.

Reply:
We could break it up into a separate workflow to run only on /release branches if that's preferable. The added cost of running it each time isn't too high either though, and it could be useful for a periodic sanity test (presumably we won't be committing to master too often)

Reply:
We've hit this duplication on Krew and it was nasty. We ended up merging the release.yml and ci.yml (same job runs both on PRs and tags).

I'd say "periodic sanity test" isn't a great argument mostly because if/when we change something, we're now potentially testing a "broken" old release/**.yml file.

And we're supposed to update release/**.yml file only when tagging releases. So it would be good to leave release/** out of CI.

Reply:
hmm that's a good point. Yeah I'll remove the test-release then to make the tests simpler